k4e-agent-sti.sh

```diff
@@ -74,32 +128,65 @@ createrepo /var/www/html/k4e-repo/
 #-----------------------------------
 git clone https://github.com/ydayagi/r4e.git /home/builder/r4e
 cd /home/builder/r4e
-./r4e-image.sh $(hostname) edgedevice helios05.lab.eng.tlv2.redhat.com:8888 http://$(hostname)/k4e-repo/
+./r4e-image.sh -s $(hostname) -i $IMAGE_NAME -o $HTTP_API -p http://$(hostname)/k4e-repo/
 
 #----------------------------------------
 # Create ISO for image and serve via http
 #----------------------------------------
-cat << EOF > /tmp/empty.toml
-name = "empty"
-description = "Empty blueprint"
+ISO_BLUEPRINT=$(mktemp)
+cat << EOF > $ISO_BLUEPRINT
+name = "edgedevice-iso"
+description = "Empty blueprint for ISO creation"
 version = "0.0.1"
 EOF
 
-if [ "$ARCH" = "aarc64" ]; then
+if [ "$ARCH" = "aarch64" ]; then
     REF="rhel/8/aarch64/edge"
 else
     REF="rhel/8/x86_64/edge"
 fi
 
-cd /var/www/html/edgedevice
-composer-cli blueprints push /tmp/empty.toml
-composer-cli compose start-ostree --ref $REF --url http://$(hostname)/edgedevice/repo/ empty edge-installer > temp.out
-BUILD_ID=$(awk '{print $2}' temp.out)
+#-------------------------
+# Save image in ISO format
+#-------------------------
+dnf install -y jq isomd5sum genisoimage
+composer-cli blueprints push $ISO_BLUEPRINT
+BUILD_ID=$(composer-cli -j compose start-ostree --ref $REF --url http://$(hostname)/$IMAGE_NAME/repo/ edgedevice-iso edge-installer | jq '.build_id')
 echo "waiting for build $BUILD_ID to be ready..."
-while [ "$(composer-cli compose status | grep $BUILD_ID | grep -c RUNNING)" != "0" ] ; do sleep 5 ; done
-if [ "$(composer-cli compose status | grep $BUILD_ID | grep -c FINISHED)" == "0" ] ; then
+while [ $(composer-cli -j compose status  | jq -r '.[] | select( .id == '$BUILD_ID' ).status') == "RUNNING" ] ; do sleep 5 ; done
+if [ $(composer-cli -j compose status  | jq -r '.[] | select( .id == '$BUILD_ID' ).status') != "FINISHED" ] ; then
     echo "image composition failed"
     echo "check 'composer-cli compose status'"
     exit 1
 fi
 composer-cli compose image $BUILD_ID
+ISO_FILE=${BUILD_ID//\"/}-installer.iso
+
+echo "copy and edit ISO contents of $ISO_FILE"
+ISO_TMP_DIR=$(mktemp -d)
+ISO_MNT_DIR=$(mktemp -d)
+mount -o loop $ISO_FILE $ISO_MNT_DIR
+cp -ra $ISO_MNT_DIR/* $ISO_TMP_DIR
+cp -ra $ISO_MNT_DIR/.??* $ISO_TMP_DIR
+umount $ISO_MNT_DIR
+rm -rf $ISO_MNT_DIR
+
+# edit kickstart file
+if [ "$ARCH" = "aarch64" ]; then
+    OS_NAME="rhel-edge"
+    REF="rhel/8/aarch64/edge"
+else
+    OS_NAME="rhel"
+    REF="rhel/8/x86_64/edge"
+fi
+
+WWW_DIR=/var/www/html/$IMAGE_NAME
```

Comment:
What will happen if that script would be run again? 
I think we should add a generated suffix

Reply:
as mentioned above, the script itself isn't idempotent. there is some cleanup to allow re-run of it, or to provide a different value as image_name.